hoomd/pytest/test_simulation.py

```diff
@@ -197,6 +197,38 @@ def test_state_from_gsd(simulation_factory, get_snapshot,
         assert box == sim.state.box
         assert_equivalent_snapshots(snap, sim.state.snapshot)
 
+@skip_gsd
+def test_state_from_snapshot(simulation_factory, get_snapshot,
```

Comment:
It would also be nice, though not a must if this also tested that it works when the snapshot data is only passed on the root rank for MPI simulations.

Reply:
Based on our discussion today, I tried to add a test to check that only the root rank gets the data from the snapshot, but I wasn't sure how to do this exactly. I'm thinking something like using an mpi4py communicator but I don't know at which step exactly only the root rank gets the data and all the other ranks don't, so that we can do the check at this step. And I'm also having some trouble rebuilding with MPI enabled. I think it would be better if we merge this PR first. I can add the test for MPI simulations after I fix my hoomd build

Reply:
@b-butler hasn't this already done? `make_gsd_snapshot` copies the HOOMD snapshot object which does not have data on ranks >= 1. The GSD snapshots on those ranks will be invalid with `snap.particles.position` and other attributes set to `None`.